internal/datastore/postgres/common/pgx.go

```diff
@@ -64,6 +73,15 @@ func queryTuples(ctx context.Context, sqlStatement string, args []any, span trac
 			if err != nil {
 				return fmt.Errorf(errUnableToQueryTuples, fmt.Errorf("unable to fetch caveat context: %w", err))
 			}
+
+			if integrityKeyID.Valid && integrityTimestamp != nil {
```

Comment:
If we pushed validation into the datastores, we can no longer using a proxy to implement it as a common solution, as it would have no access to the integrity information